javascript/ql/src/Security/CWE-178/CaseSensitiveMiddlewarePath.ql

```diff
@@ -0,0 +1,121 @@
+/**
+ * @name Case-sensitive middleware path
+ * @description Middleware with case-sensitive paths do not protect endpoints with case-insensitive paths
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 7.3
+ * @precision high
+ * @id js/case-sensitive-middleware-path
+ * @tags security
+ *       external/cwe/cwe-178
+ */
+
+import javascript
+
+/**
+ * Converts `s` to upper case, or to lower-case if it was already upper case.
+ */
+bindingset[s]
+string invertCase(string s) {
+  if s.regexpMatch(".*[a-z].*") then result = s.toUpperCase() else result = s.toLowerCase()
+}
+
+RegExpCharacterClass getEnclosingClass(RegExpTerm term) {
+  term = result.getAChild()
+  or
+  term = result.getAChild().(RegExpRange).getAChild()
+}
+
+/**
+ * Holds if `term` distinguishes between upper and lower case letters, assuming the `i` flag is not present.
```

Comment:
I agree that this predicate suits the query: it identifies the typical regexp routes such as `/admin\/.*/`.
The predicate will however also hold (wrongly) for cases such as `/a|A/` or `/[abcA-C]/`

Could the docstring/predicate name be something like `isCaseSensitiveRouteRegExp`?

Reply:
Renamed to `isLikelyCaseSensitiveRegExp`